### ISSUE TITLE
[Documentation] Update INSTALL.md to remove redundancy and make a clearer set of high-level steps and create README-CNF_Setup.md

### ISSUE THREAD

**Issue by taylor** [cncf/cnf-conformance#526]
**Is your feature request related to a problem? Please describe.**
- Confusing mix of source, binary, and development commands
- Redundancy of commands

Right now the https://github.com/cncf/cnf-conformance/blob/master/INSTALL.md guide is hard to follow if you are not already familiar with source and binary installation and setup for the test suite. If someone wants a binary based install its not clear when and where to move past other instructions.

There is redundancy of commands to run throughout the INSTALL guide such as when to run the `setup` command (at least 18 times at the time of the issue creation).

There are commands listed under install that are really configuration of the test suite. There are instructions for running commands which are for testing a CNF before the configuration section.

There is extensive information on configuring a CNF in multiple places in the documentation and other documents.

General usage is repeated and not consistent across the document and with other documents.

**Describe the solution you'd like**
- Create a TOC for the INSTALL doc
- Items for Install guide
   -  Prereqs
      - Common
      - Pre-built binary
      - Source built binary
   - How to obtaining a binary either pre-built or building from source
   - Command(s) to prepare the test suite eg. `setup`
   - Configuring CNF to be tested
   - Minimal set of commands running the test suite to test your CNF
      - Link to USAGE for additional testing commands
- Move more extensive developer instructions to a new document, or at least hidden collapsable sections
- Remove redundancy such as usage instructions in Binary Install, Dev Install, Setup, CNF Configuration, etc
- Add prereq collapsable section for source prerqs 
- Simplify build binary from source
   - Choose build manually or with with docker.  Move the rest to a dev doc.
   - For manual, choose normal or static release.  Pick the one most likely to succeed (usually no options). Hide the other choice.
   - If docker is used only have prereqs for docker and move the rest to a dev page
- Cleanup CNF Configuration / setup
   - Provide link to a document focused on configuration of CNF 
   -  Create collapsed section for downloading CoreDNS example with wget
- Move most CNF configuration instructions to a new doc: ex. README-CNF_configuration.md
   - move `Overview (for setting up your own cnf)`
   - move configuration content from `Detailed Steps (for setting up your own cnf)`
   - merge https://github.com/cncf/cnf-conformance/blob/master/CNF_CONFORMANCE_YML_USAGE.md for additional details
- Keep ready to rock :)
- Minimal commands to test - In https://github.com/cncf/cnf-conformance/blob/master/INSTALL.md#running-and-checking-results-for-the-conformance-testing
   - Update `Running all (workload and platform) tests` section to 
      - use cnf_setup at the start
      - keep all, workload, and platform
      - add link to USAGE.md
- Move More Example usage or delete competely
- Move bottom dev section to new Dev document

**Describe alternatives you've considered**
- Split everything into their own docs
- Deleting all but *installation instructions*

**Additional context**

<img width="950" alt="Screen Shot 2020-12-10 at 5 16 02 PM" src="https://user-images.githubusercontent.com/26697/101841433-88c19880-3b0b-11eb-9d4f-feb7512f5187.png">

<img width="1108" alt="Screen Shot 2020-12-10 at 5 14 42 PM" src="https://user-images.githubusercontent.com/26697/101841301-47c98400-3b0b-11eb-9eee-079b73f6c8bc.png">

**Comment by agentpoyo** [comment]
FYI, 

Using hackmd so I can view changes in real time. You can see my draft before we create branch to copy/paste/replace at https://hackmd.io/6h7NXdHnR4qUYgnnQPy5UA

Focusing on the main INSTALL.md for now whlie @williscool is doing the developer split off doc.

@williscool @taylor 

Feel free to make contributions/edits, etc if you feel obligated to do so while it's a WIP for the overhaul.

**Comment by williscool** [comment]
Started on source install here: I think the 2 should closely mirror each other as much as possible but also not have irrelavant parts or examples (I removed everything related to curl and binary install from the new source install doc)

But i think they should both have a TOC and at least a 2 or 3 examples (a lot of people won't click through to see other usage docs without a little taste. they might just bail out altogether and think its too complicated)

https://hackmd.io/T0XVcwF2TFSvWrSmF966FA

**Comment by taylor** [comment]
@williscool, @agentpoyo the end section [Running and checking results for the Conformance testing
](https://github.com/cncf/cnf-conformance/blob/master/INSTALL.md#running-and-checking-results-for-the-conformance-testing) of the current install has a good section for showing a few examples:

<img width="907" alt="Screen Shot 2020-12-16 at 10 11 13 AM" src="https://user-images.githubusercontent.com/26697/102374643-33600e00-3f87-11eb-9626-652f23a9a8be.png">

After its updated with correct information (eg. not passing options to all as those will be shown with the cnf_setup command) a few other things can be shown, such as running a category

![image](https://user-images.githubusercontent.com/26697/102374917-7fab4e00-3f87-11eb-87f2-c2bd6de14142.png)

and running a single test like

![image](https://user-images.githubusercontent.com/26697/102375043-a5385780-3f87-11eb-86a9-2c73b6277b7f.png)

Then a link to the usage page for more.

I don't think the test command usage should be shown again and again throughout the INSTALL or SOURCE-INSTALL document like it is now. 

Kill the duplicates (individual items and entire sections), such as

<img width="1036" alt="Screen Shot 2020-12-16 at 10 20 16 AM" src="https://user-images.githubusercontent.com/26697/102375793-88e8ea80-3f88-11eb-9f53-f37173ebdd8f.png">


Give the command immediately relevant to where you are now (eg. I'm configuring the CNF so show the cnf_setup command to use with the cnf-conformance.yml)


![image](https://user-images.githubusercontent.com/26697/102375887-a74ee600-3f88-11eb-8be9-a857f3cae8f5.png)

**Comment by agentpoyo** [comment]
> I think the 2 should closely mirror each other as much as possible but also not have irrelavant parts or examples (I removed everything related to curl and binary install from the new source install doc)

Yeah, I'm wondering if we could strip it down more though, like the common steps of setting up k8s, perhaps a reference or link back to where it's located in the main INSTALL as well.

Once we get both separate, we can dissect further perhaps to what makes most sense.
